Eradication/VectorPopulationIndividual.cpp

```diff
@@ -865,6 +870,18 @@ namespace Kernel
         for( auto it = pAdultQueues->begin(); it != pAdultQueues->end(); ++it )
         {
             adapter.SetVectorID( ( *it )->GetID() );
+            if( m_pMigrationInfoVector->GetMigrationAlleleCombinationsSize() > 1 )
+            {
+                // only do this if we actually have by gene migration happening
+                VectorGenome vc_genome = ( *it )->GetGenome();
+                int migration_data_index = GetMigrationDataIndex( vc_genome );
+                m_pMigrationInfoVector->SetIndFemaleRates( migration_data_index ); // updating m_TotalRateFemale and m_RateCDFFemale for these genes
+                if( m_pMigrationInfoVector->GetTotalRate( m_pMigrationInfoVector->ConvertVectorGender( vc_genome.GetGender() ) ) == 0 )
+                {
+                    continue;
+                }
+            }
```

Comment:
Maybe we use GetTravelData( genome ) here as well as in VP.  It would return the reachable nodes and fraction traveling, but all we would use it for is to check if the pointer is null.  GetTravelData() could also do the same job as SetIndFemaleRates().  The only slightly annoying thig is that GetTravelData() can be const if we do this.  I could live with that.

One alternative that I started looking at was changing the interface to IMigrationInfo so that PickMigrationStep() could pass the traveler or something so that we could somehow get the vector genome, but maybe the SetIndFemaleRates()/GetTravelData() functionality is fine.  We just need to have a comment here telling the reader that it is setting up data for PickMigrationStep()

Reply:
I'm using "GetTravelingFraction" to do all that.